src/User/Controller/AuthController.php

```diff
@@ -107,12 +107,11 @@ public function register(Request $request, Response $response): Response
         $parsedData = $request->getParsedBody();
 
         $this->validationService->validate($parsedData, [
-            UserInterface::COLUMN_USERNAME => 'required|min:2|max:12|regex:/^[a-zA-Z\d]+$/',
+            UserInterface::COLUMN_USERNAME => 'required|min:3|max:12|regex:/^[a-zA-Z]+([\.:_\-=!\?]){0,5}([a-zA-Z0-9])+$/',

```

Comment:
What's the reason for this additional regex check ?

Reply:
Actually it would be cool to be able to add your own custom regular expression through configuration so that Owners/Developers can register Usernames as the want. In my case I wanted the possibility to add _.:_-=!_ characters in between letters.